src/sentry/shared_integrations/client/base.py

```diff
@@ -78,6 +85,40 @@ def finalize_request(self, prepared_request: PreparedRequest) -> PreparedRequest
         """
         return prepared_request
 
+    def _get_redis_key(self):
+        """
+        Returns the redis key for the integration or empty str if cannot make key
+        """
+        if not hasattr(self, "integration_id"):
+            return ""
+        if not self.integration_id:
+            return ""
+        return f"sentry-integration-error:{self.integration_id}"
+
+    def is_error(self, e: Exception) -> bool:
+        if type(e) is ConnectionError:
+            return True
+        if type(e) is Timeout:
+            return True
+        if type(e) is HTTPError:
+            return True
+
+        return False
+
+    def is_response_fatal(self, resp: Response) -> bool:
+        return False
+
+    def is_response_error(self, resp: Response) -> bool:
+        if resp.status_code:
+            if resp.status_code >= 400 and resp.status_code != 429 and resp.status_code < 500:
+                return True
+        return False
+
+    def is_response_success(self, resp: Response) -> bool:
+        if resp.status_code < 300:
```

Comment:
Looks like there's a [test failure](https://github.com/getsentry/sentry/actions/runs/5616794974/job/15219775331?pr=52994) here so just check that there's a status code first before comparing it like you do in `is_response_error`.